lib/manager/gradle-lite/parser.ts

```diff
@@ -0,0 +1,294 @@
+      tokenMap,
+      variables,
+    }: SyntaxHandlerInput): SyntaxHandlerOutput {
+      const token = tokenMap.depInterpolation as StringInterpolation;
+      const interpolationResult = interpolateString(token.children, variables);
+      if (interpolationResult && isDependencyString(interpolationResult)) {
+        const dep = parseDependencyString(interpolationResult);
+        if (dep) {
+          const lastChild = token.children[token.children.length - 1];
+          const lastChildValue = lastChild?.value;
+          const variable = variables[lastChildValue];
+          if (
+            lastChild?.type === TokenType.Variable &&
+            variable &&
+            variable?.value === dep.currentValue
+          ) {
+            dep.managerData = {
+              fileReplacePosition: variable.fileReplacePosition,
+              packageFile: variable.packageFile,
+            };
+            dep.groupName = variable.key;
+            return { deps: [dep] };
+          }
+        }
+      }
+      return null;
+    },
+  },
+  {
+    // id 'foo.bar' version '1.2.3'
+    matchers: [
+      { matchType: TokenType.Word, matchValue: 'id' },
+      { matchType: TokenType.String, tokenMapKey: 'pluginName' },
+      { matchType: TokenType.Word, matchValue: 'version' },
+      { matchType: TokenType.String, tokenMapKey: 'pluginVersion' },
+    ],
+    handler: function processPlugin({
+      tokenMap,
+      packageFile,
+    }: SyntaxHandlerInput): SyntaxHandlerOutput {
+      const { pluginName, pluginVersion } = tokenMap;
+      const dep = {
+        depType: 'plugin',
+        depName: pluginName.value,
+        lookupName: `${pluginName.value}:${pluginName.value}.gradle.plugin`,
+        registryUrls: ['https://plugins.gradle.org/m2/'],
+        currentValue: pluginVersion.value,
+        commitMessageTopic: `plugin ${pluginName.value}`,
+        managerData: {
+          fileReplacePosition: pluginVersion.offset,
+          packageFile,
+        },
+      };
+      return { deps: [dep] };
+    },
+  },
+];
+
+interface MatchConfig {
+  tokens: Token[];
+  variables: PackageVariables;
+  packageFile: string;
+}
+
+function tryMatch({
+  tokens,
+  variables,
+  packageFile,
+}: MatchConfig): SyntaxHandlerOutput {
+  for (const { matchers, handler } of matcherConfigs) {
+    const tokenMap = matchTokens(tokens, matchers);
+    if (tokenMap) {
+      const result = handler({
+        packageFile,
+        variables,
+        tokenMap,
+      });
+      if (result !== null) {
+        return result;
+      }
+    }
+  }
+  tokens.shift();
+  return null;
+}
+
+export function parseGradle(
+  input: string,
+  initVars: PackageVariables = {},
+  packageFile?: string
+): PackageDependency<ManagerData>[] {
+  logger.trace({ packageFile }, `Gradle parsing ${packageFile} start`);
+  const variables = { ...initVars };
+  const dependencies: PackageDependency<ManagerData>[] = [];
+
+  const startTime = Date.now();
```

Comment:
I would use something like [`perf_hooks`](https://nodejs.org/api/perf_hooks.html) or [`process.hrtime`](https://nodejs.org/api/process.html#process_process_hrtime_bigint). `Date` objects do have 1ms precision, but they're not intended to be used for performance monitoring.